contracts/HoprStake2.sol

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC777/IERC777Recipient.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC1820Registry.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "./IHoprBoost.sol";
+
+/**
+ * 
+ */
+contract HoprStake2 is Ownable, IERC777Recipient, IERC721Receiver, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using Math for uint256;
+
+    struct Account {
+        uint256 actualLockedTokenAmount; // The amount of LOCK_TOKEN being actually locked to the contract. 
+                                         // Those tokens can be withdrawn after “UNLOCK_START”
+        uint256 lastSyncTimestamp; // Timestamp at which any “Account” attribute gets synced for the last time. 
+        uint256 cumulatedRewards; // Rewards accredited to the account at “lastSyncTimestamp”.
+        uint256 claimedRewards; // Rewards claimed by the account.
+    }
+
+    uint256 public constant PROGRAM_START = 1642424400; // Block timestamp at which incentive program starts. Default value is 1642424400 (Jan 17th 2022 14:00 CET).
+    uint256 public constant PROGRAM_END = 1650974400; // Block timestamp at which incentive program ends. From this timestamp on, tokens can be unlocked. Default value is 1650974400 (Apr 26th 2022 14:00 CET).
```

Comment:
comment about timestamp seems to be wrong, 26 April 2022 13:00 CET

Reply:
1650974400 is `Tue Apr 26 2022 14:00:00 GMT+0200 (Central European Summer Time)` It's 2 pm summer time.